pkg/api/nextdate.go

```diff
@@ -0,0 +1,242 @@
+	switch splitRepeat[0] {
+	case "d":
+		return dailyRepeat(now, date, splitRepeat)
+	case "y":
+		return yearlyRepeat(now, date)
+	case "w":
+		return weeklyRepeat(now, date, splitRepeat)
+	case "m":
+		return monthlyRepeat(now, date, splitRepeat)
+	default:
+		return "", fmt.Errorf("invalid repeat format: %v", splitRepeat[0])
+	}
+}
+
+// Функция dailyRepeat добавляет указанное кол-во дней
+func dailyRepeat(now, date time.Time, splitRepeat []string) (string, error) {
+	if len(splitRepeat) != 2 {
+		return "", errInvalidRepeat
+	}
+
+	// Проверки на правильный формат дней
+	days, err := strconv.Atoi(splitRepeat[1])
+	if err != nil {
+		return "", fmt.Errorf("invalid days format: %v", err)
+	}
+	if days > 400 {
+		return "", fmt.Errorf("the maximum allowed interval (%v days) has been exceeded", 400)
+	}
+	if days <= 0 {
+		return "", fmt.Errorf("days value must be positive: %v", days)
+	}
+
+	for {
+		date = date.AddDate(0, 0, days)
+		if afterNow(date, now) {
+			break
+		}
+	}
+
+	return date.Format(dateFormat), nil
+}
+
+// Функция yearlyRepeat добавляет 1 год к дате
+func yearlyRepeat(now, date time.Time) (string, error) {
+	for {
+		date = date.AddDate(1, 0, 0)
+		if afterNow(date, now) {
+			break
+		}
+	}
+
+	return date.Format(dateFormat), nil
+}
+
+// Сложные функции...
+// Функция weeklyRepeat переносит задачи на указанные дни недели
+func weeklyRepeat(now, date time.Time, splitRepeat []string) (string, error) {
+	if len(splitRepeat) != 2 {
+		return "", errInvalidRepeat
+	}
+
+	weekDays := strings.Split(splitRepeat[1], ",")
+	//Создаем map для хранения дней
+	days := make(map[int]bool, len(weekDays))
+
+	for _, v := range weekDays {
+		v, err := strconv.Atoi(v)
+		if err != nil || v < 1 || v > 7 {
+			return "", fmt.Errorf("invalid weekday: %v", v)
+		}
+		days[v] = true
+	}
+
+	// Начинаем со следующего дня
+	date = date.AddDate(0, 0, 1)
+
+	for {
+		weekDay := int(date.Weekday())
+		if weekDay == 0 {
+			weekDay = 7
+		}
+
+		if days[weekDay] && afterNow(date, now) {
+			break
+		}
+
+		date = date.AddDate(0, 0, 1)
+	}
+
+	return date.Format(dateFormat), nil
+}
+
+// Функция monthlyRepeat переносит задачи на указанные дни месяца
+func monthlyRepeat(now, date time.Time, splitRepeat []string) (string, error) {
+	// Проверка качества (:
+	if len(splitRepeat) < 2 {
+		return "", errInvalidRepeat
+	}
+
+	// Добавляем дни в slice
+	daySpl := strings.Split(splitRepeat[1], ",")
+	days := make([]int, 0, len(daySpl))
+	for _, v := range daySpl {
+		day, err := strconv.Atoi(v)
+		if err != nil || day < -2 || day == 0 || day > 31 {
+			return "", fmt.Errorf("invalid day format: %v", day)
+		}
+		days = append(days, day)
+	}
+
+	// Парсим месяцы, если указаны
+	months := make(map[int]bool)
+	if len(splitRepeat) > 2 {
+		monthParts := strings.Split(splitRepeat[2], ",")
+		for _, m := range monthParts {
+			month, err := strconv.Atoi(m)
+			if err != nil || month < 1 || month > 12 {
+				return "", fmt.Errorf("invalid month format: %v", month)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
				return "", fmt.Errorf("invalid month format: %w", month)
```